### PR TITLE
Update CI check job to ensure all platforms

### PR DESCRIPTION
**Pull request by yvan-sraka**

### CODE REVIEW

**Comment by yvan-sraka** [comment]
I could also add that in `test` / `doctests` jobs if we want to be sure for e.g. that a test will not succeed on Linux and fail on Windows :)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -29,7 +29,13 @@ jobs:
 
   check:
     needs: sanity
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
```
Comment (adrien-zinger) [inline .github/workflows/ci.yml:36]: According to the [documentation](https://docs.gitlab.com/ee/ci/yaml/#rulesif) it's possible to target a branch.

Should we build on every platforms only when we merge into main or stage?

Comment (yvan-sraka) [reply .github/workflows/ci.yml:36]: Yes, you right! I will enable multiple platforms for `check`, `test` and `doctest` jobs only on `main` and `stage` branch :)

**Comment by yvan-sraka** [comment]
If we start adding cargo features it could also be nice to add https://github.com/taiki-e/cargo-hack

**Comment by yvan-sraka** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors cancel

**Comment by yvan-sraka** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors cancel

**Comment by yvan-sraka** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors cancel

**Comment by yvan-sraka** [comment]
bors merge

**Pull request merged**